### ISSUE TITLE
[BUG] [Linter] Unbalanced block tag should raise error

### ISSUE THREAD

**Issue by allcaps** [djlint/djLint#1177]
- [x] I'm on the [latest version](https://pypi.org/project/djlint/) of djLint
- [x] I've searched the [issues](https://github.com/djlint/djLint/issues)
- [x] I've read the [docs](https://djlint.com)

## System Info

- OSX
- Python Version 3.12.9
- djLint Version 1.36.4
- template language: django

## Issue

Unbalanced block tags go unnoticed

## How To Reproduce

Create a template:

```
{% block foo %}
{% endblock bar %}
```
Run `djlint . --lint --check`.
